### PR TITLE
Check if protected first

### PR DESCRIPTION
**Pull request by jacobweinstock**
Updated `doProtected` to check if the method should be protected by auth first and then grab the token if the method should be protected. Also added method comments to explain this.

### CODE REVIEW

**Review by jgavinray — DISMISSED** [review]
LGTM - thanks for putting this together Jacob!  I appreciate the defensive programming on this one.

File: grpc/authz/authz.go [hunk 1]
```diff
@@ -121,19 +121,22 @@ func permissionDeniedError(msg string) error {
 	return status.Errorf(codes.PermissionDenied, "no permission to access this RPC %s", msg)
 }
 
+// doProtected checks if the method should be protected by auth or not. If so, then scopes
+// for the method are saved to Config.scopes for later use.
```
[the remark this answers is not in the document]
Comment (jacobweinstock) [reply grpc/authz/authz.go:125]: something like this? https://github.com/packethost/pkg/blob/fb44dce23cb3ce48e0412c725951f412a6d6ad70/grpc/authz/authz.go#L24

**Review by jgavinray — APPROVED** [review]

File: grpc/authz/authz.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmlb) [reply grpc/authz/authz.go:123]: Sort of, like:
`doProtected checks if the method should be protected by auth or not by checking if the method name is present in c.ScopeMapping`.

But then the ScopeMapping doc also doesn't help me understand what its going to be used for either. Is it just me? Should I know that scope mapping maps rpc method names to protection requirement?

[the remark this answers is not in the document]
Comment (jacobweinstock) [reply grpc/authz/authz.go:123]: I pushed a new commit updating the `Config.ScopeMapping` comment.
https://github.com/packethost/pkg/pull/38/files#diff-a1f2267dc6b1025ebd47ec4811d69c0125e95952579fecc39af9a706de267770R24-R27

[the remark this answers is not in the document]
Comment (mmlb) [reply grpc/authz/authz.go:123]: thanks that helps now :D

**Pull request merged**